gcn/notices/fermi/gbm/strict/alert.schema.json

```diff
@@ -0,0 +1,168 @@
+{
+  "$id": "https://gcn.nasa.gov/schema/main/gcn/notices/fermi/gbm/strict/alert.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "GCN Schema for Fermi GBM Alert Notices",
+  "description": "This schema is a stricter version of the more permissive main schema, https://gcn.nasa.gov/schema/main/gcn/notices/fermi/gbm/alert.schema.json.  The Kafka server uses the main schema for new notices.  Each Alert notice (classic numeric type 110) must validate against the main schema and should also validate against this one.  Examples in this directory come in two versions referencing the main and the strict schemas respectively.  The intent of the strict schema is to help understanding the Alert notice, since it better describes what content to be expected.",
```

Comment:
I still don't understand the purpose or use of this strict schema.  Each schema is a Kafka topic, though you are welcome to multiple examples.  The GBM topic will be gcn.notices.fermi.gbm.alert, so what is the use case for gcn.notices.fermi.gbm.strict.alert?

Reply:
I wasn't planning on having a `gcn.notices.fermi.gbm.strict.alert` topic, only the `gcn.notices.fermi.gbm.alert`. The same way the server functionality doesn't depend on the examples, it should ignore the strict schemas. If the server does notice validation, it should use only the `.../gbm/alert.schema.json`. All I want the strict schema to do, is will just be there in addition to the examples and will serve a similar illustrative/documentation purpose.

The JSON objects described by the strict schema are a subset of the objects described by the "non-sctrict" one. If, by looking at the examples, one has doubts about what to expect, then the strict schema will help clear it up. This will become more so with the other notice types.

Reply:
@shb46, I looked at a diff between gcn/notices/fermi/gbm/alert.schema.json and gcn/notices/fermi/gbm/strict/alert.schema.json. It looks like the only difference is that in the former, you include types from core schema, whereas in the latter, you hardcode the definitions of the fields from the core schema that you are actually using. Please don't do that! Please remove the "strict" schema directory from your PR.

I think that this exposes an interesting issue, that when you include a core schema, it is not clear to users which of its many optional fields you actually intend to populate. That affects all GCN notice types, not just Fermi. @jracusin, let's open another issue to study that problem and come up with options.

Reply:
Done.